docs/Guides/Testing.md

```diff
@@ -243,33 +243,59 @@ after initializing routes and plugins with `fastify.ready()`.
 
 Uses **app.js** from the previous example.
 
-**test-listen.js** (testing with
-[`Request`](https://www.npmjs.com/package/request))
+**test-undici.js** (testing with [`undici`](https://www.npmjs.com/package/undici))
 ```js
 const tap = require('tap')
-const request = require('request')
+const { Client } = require('undici')
 const buildFastify = require('./app')
 
-tap.test('GET `/` route', t => {
+tap.test('should work with undici', async t => {
   t.plan(5)
 
   const fastify = buildFastify()
 
   t.teardown(() => fastify.close())
 
-  fastify.listen({ port: 0 }, (err) => {
-    t.error(err)
+  await fastify.listen()
 
-    request({
-      method: 'GET',
-      url: 'http://localhost:' + fastify.server.address().port
-    }, (err, response, body) => {
-      t.error(err)
-      t.equal(response.statusCode, 200)
-      t.equal(response.headers['content-type'], 'application/json; charset=utf-8')
-      t.same(JSON.parse(body), { hello: 'world' })
-    })
-  })
+  const client = new Client(
+    'http://localhost:' +
+    fastify.server.address().port
+  )
+
+  const response = await client.request({ method: 'GET', path: '/' })
+
+  t.equal(await response.body.text(), 'Hello from fastify-slow-down!')
+  t.equal(response.statusCode, 200)
+})
+```
+**test-fetch.js** (testing with
+[`fetch`](https://nodejs.org/docs/latest-v18.x/api/globals.html#fetch))
+> **_NOTE:_** `fetch` is available in Node 18
+```js
+const tap = require('tap')
+const buildFastify = require('./app')
+
+tap.test('should work with fetch', async t => {
+  t.plan(5)
+
+  const fastify = buildFastify()
+
+  t.teardown(() => fastify.close())
+
+  await fastify.listen()
+  
+  const response = await fetch(
+    'http://localhost:' +
+    fastify.server.address().port
+  )
+
+  t.equal(response.status, 200)
+  t.equal(
+    response.headers.get('content-type'),
+    'application/json; charset=utf-8'
+  )
+  t.has(await response.json(), { hello: 'world' })
 })
 ```
```

Comment:
Why duplicate the example?

Reply:
@jsumners is not a duplicate, the first one is an example using undici and the second example using fetch from Node 18

Reply:
They do exactly the same thing. Thus, they are duplicate examples. Are we trying to say fetch is so different that someone using it can't understand the basic client example? It doesn't look so to me from the provided examples.

Reply:
That is not clear in the documentation as it is presented. It just seems like a second snippet that is very slightly different for no discernible reason.

Reply:
I will add between the `test-undici` and `test-fetch` a section `Another example`, the readers will understand that is the same approach but is using another package